app/redux/smesher/actions.ts

```diff
@@ -86,3 +87,32 @@ export const resumeSmeshing = () => async (
     ? dispatch(startSmeshing(smeshingOpts))
     : false;
 };
+
+export const updatePostProvingOpts = (
+  nonces: number,
+  threads: number
+) => async (_, getState: GetState) =>
+  eventsService.updateProvingOpts(getGenesisID(getState()), {
+    nonces,
+    threads,
+  });
+
+export const updateProvingOptsAndRestartSmeshing = (
+  nonces: number,
+  threads: number,
+  postSetupState: PostSetupState,
+  isSmeshingStarted: boolean
+) => async (dispatch: AppThDispatch) => {
+  // update node-config file
+  await dispatch(updatePostProvingOpts(nonces, threads));
+
+  // resolve an issue with Windows, when we use to much resources RAM and CPU and smeshing was stopped in config
+  // if in config smeshing ['smeshing-start'] === true
+  if (isSmeshingStarted) {
+    // if smeshing in progress postSetupState === PostSetupState.STATE_IN_PROGRESS
+    postSetupState === PostSetupState.STATE_IN_PROGRESS &&
+      (await dispatch(pauseSmeshing()));
+    // handle start smeshing, PostSetupState.STATE_PAUSED or STATE_NOT_STARTED
+    await dispatch(resumeSmeshing());
```

Comment:
Can you describe better what is the logic behind this?

You're pausing smeshing only if it was in the middle of the creation of PoS data (but not when it is already finished creating, for example).
And trying to resume smeshing all the time (which has internal checks for some redux state variables). Why?

So it may work somehow, but I'd prefer to have one of these options:

1. Main process handle `eventsService.updateProvingOpts` properly, including restarting the node if needed
2. Make it clear why and when we need to call them, double-check that all variables are actual and not stuck somewhere. Basically, I don't like that we rely on some state in the GUI :)